src/main/java/org/apache/sling/feature/cpconverter/handlers/BundleEntryHandler.java

```diff
@@ -109,111 +154,309 @@ public void handle(@NotNull String path, @NotNull Archive archive, @NotNull Entr
+        String repositoryPath = (pathEntry.get().getTarget() != null ? pathEntry.get().getTarget() : "/") + URLDecoder.decode(entryName.substring(pathEntry.get().getPath().length()), "UTF-8");
+        String contentPackagePath = org.apache.jackrabbit.vault.util.Constants.ROOT_DIR + PlatformNameFormat.getPlatformPath(repositoryPath);
+
+        // in which content package should this end up?
+        VaultPackageAssembler packageAssembler = initPackageAssemblerForPath(bundleArtifactId, repositoryPath, pathEntry.get(), packageAssemblers, converter);
+        Path tmpInputFile = null;
+        if (contentParserAndOptions != null) {
+            // convert to docview xml
+            tmpInputFile = Files.createTempFile(converter.getTempDirectory().toPath(), "docview", ".xml");
+            try (OutputStream docViewOutput = Files.newOutputStream(tmpInputFile, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
+                 DocViewSerializerContentHandler contentHandler = new DocViewSerializerContentHandler(docViewOutput, nsRegistry)) {
+                contentParserAndOptions.getKey().parse(contentHandler, bundleFileInputStream, contentParserAndOptions.getValue());
+                contentPackagePath = FilenameUtils.removeExtension(contentPackagePath) + ".xml";
+            } catch (IOException e) {
+                throw new IOException("Can not parse " + jarEntry, e);
+            } catch (DocViewSerializerContentHandlerException e) {
+                throw new IOException("Can not convert " + jarEntry + " to enhanced DocView format", e);
+            }
+        }
 
-                Objects.requireNonNull(converter.getArtifactsDeployer()).deploy(new InputStreamArtifactWriter(input), id);
+        // does entry in initial content need to be extracted into feature model (e.g. for OSGi configurations)?
+        EntryHandler entryHandler = converter.getHandlersManager().getEntryHandlerByEntryPath(contentPackagePath);
+        if (entryHandler != null) {
+            if (tmpInputFile == null) {
+                tmpInputFile = Files.createTempFile(converter.getTempDirectory().toPath(), "initial-content", Text.getName(jarEntry.getName()));
+                try (OutputStream tmpBundleOutput = Files.newOutputStream(tmpInputFile, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING)) {
+                    IOUtils.copy(bundleFileInputStream, tmpBundleOutput);
+                }
+            }
+            // remap CND files to make sure they are picked up by NodeTypesEntryHandler
+            if (nsRegistry.getRegisteredCndSystemIds().contains(jarEntry.getName())) {
+                contentPackagePath = "/META-INF/vault/" + Text.getName(jarEntry.getName()) + ".cnd";
+            }
+            try (SingleFileArchive archive = new SingleFileArchive(tmpInputFile.toFile(), contentPackagePath)) {
+                entryHandler.handle(repositoryPath, archive, archive.getRoot(), converter);
```

Comment:
Additionally, when I fix the contentpackage path above, it finds the right handler, but then I think it might be wrong to use the repositoryPath here - that one will not match the handler.

Reply:
The handler should get `contentPackagePath` instead of `repositoryPath`. Gonna fix this.

Reply:
Fixed in https://github.com/apache/sling-org-apache-sling-feature-cpconverter/pull/74/commits/9a14efcb40289a6c14b1516cc9bb4f40fc187588.